From dec014cd019ce4ccf6b1e7579ff4d58ec54e33f8 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 1 Jun 2017 14:34:10 +0800
Subject: [PATCH 48/65] fix installed system boot from encrypt fs failed

While using cryptsetup to encrypt filesystem, it failed to boot
installed system.
-----------------
dracut: luksOpen /dev/sda3 luks-9cf2d2d9-849e-4ecb-a7f8-5918259306d6
Enter passphrase for /dev/sda3: Switched to clocksource tsc

device-mapper: table: 253:0: crypt: Error allocating crypto tfm
device-mapper: ioctl: error adding target to table
device-mapper: reload ioctl on  failed: No such file or directory
Failed to setup dm-crypt key mapping for device /dev/sda3.
Check that kernel supports aes-xts-plain64 cipher (check syslog for more info).
Wrong password
-----------------

The reason is the initramfs (generated by dracut) missed kernel module.

Config dracut to let initramfs has the kernel driver.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/payload/__init__.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pyanaconda/payload/__init__.py b/pyanaconda/payload/__init__.py
index d29fa62..00ff79f 100644
--- a/pyanaconda/payload/__init__.py
+++ b/pyanaconda/payload/__init__.py
@@ -672,6 +672,10 @@ class Payload(object):
         for subdir in ["log", "tmp"]:
             iutil.mkdirChain(iutil.getSysroot() + "/var/volatile/%s" % subdir)
 
+        with open(iutil.getSysroot() + "/etc/dracut.conf.d/cryptsetup.conf", "w") as f:
+            f.write('add_drivers+=" aesni_intel"\n')
+            f.write('add_drivers+=" aes_x86_64"\n')
+
         for kernel in self.kernelVersionList:
             log.info("recreating initrd for %s", kernel)
             if not flags.imageInstall:
-- 
2.7.4

